File: 1/Android Básico/CalculaIMC/app/src/main/java/br/edu/utfpr/calculaimc/MainActivity.java
package br.edu.utfpr.calculaimc;

import android.graphics.Color;
import android.os.Bundle;
import android.util.Log;
import android.view.KeyEvent;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.text.DecimalFormat;

public class MainActivity extends AppCompatActivity {

    private EditText  etPeso;
    private EditText  etAltura;
    private TextView  tvResultado;
    private Button    btCalcular;
    private Button    btLimpar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etPeso = findViewById( R.id.etPeso );
        etAltura = findViewById( R.id.etAltura );
        tvResultado = findViewById( R.id.tvResultado );
        btCalcular = findViewById( R.id.btCalcular );
        btLimpar = findViewById( R.id.btLimpar );

        btCalcular.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                btCalcularOnClick();
            }
        });

        btLimpar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                btLimparOnClick();
            }
        });

        etAltura.setOnEditorActionListener( new TextView.OnEditorActionListener() {
            @Override
            public boolean onEditorAction( TextView textView , int i , KeyEvent keyEvent ) {
                btCalcularOnClick();
                return false;
            }
        });
    }

    private void btCalcularOnClick() {

        if ( etPeso.getText().toString().isEmpty() ){
            etPeso.setError("Campo Peso deve ser preenchido.");
            etPeso.requestFocus();
            return;
        }
        if ( etAltura.getText().toString().isEmpty() ){
            etAltura.setError("Campo Altura deve ser preenchido.");
            etAltura.requestFocus();
            return;
        }

        double peso = Double.parseDouble(etPeso.getText().toString());
        double altura = Double.parseDouble(etAltura.getText().toString());

        double resultado = peso / Math.pow( altura, 2 );

        DecimalFormat df = new DecimalFormat( "0.00" );
        tvResultado.setText( df.format( resultado) );

        if( resultado < 18.5 ){
            tvResultado.setTextColor(Color.BLUE);
        }else if( resultado > 18.5 && resultado < 24.9 ){
            tvResultado.setTextColor(Color.GREEN);
        }else if( resultado > 25 && resultado < 29.9 ){
            tvResultado.setTextColor(Color.rgb(255,210,0));
        }else if( resultado > 30 && resultado < 34.9 ){
            tvResultado.setTextColor(Color.rgb(255,165,0));
        }else if( resultado >= 35 ){
            tvResultado.setTextColor(Color.RED);
        }


    }

    private void btLimparOnClick(){
        etPeso.setText( "" );
        etAltura.setText( "" );
        tvResultado.setText( "0.0" );
        tvResultado.setTextColor(Color.BLACK);
        etPeso.requestFocus();
    }
}

File: 1/Android Aplicado/UsandoComponentesVisuais/app/src/main/java/br/edu/utfpr/usandocomponentesvisuais/MainActivity.kt
package br.edu.utfpr.usandocomponentesvisuais

import android.annotation.SuppressLint
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.LinearLayout
import android.widget.RadioButton
import android.widget.RadioGroup
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.google.android.material.snackbar.Snackbar

class MainActivity : AppCompatActivity() {

    private lateinit var rgSexo: RadioGroup

    private lateinit var rbMasculino: RadioButton
    private lateinit var rbFeminino: RadioButton
    private lateinit var main : LinearLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        rgSexo = findViewById(R.id.rgSexo)
        rbMasculino = findViewById(R.id.rbMasculino)
        rbFeminino = findViewById(R.id.rbFeminino)

        main = findViewById(R.id.main)

        rgSexo.setOnCheckedChangeListener{ _, checkedId ->
            Snackbar.make(main , "Componente Selecionado", Snackbar.LENGTH_SHORT).show()
        }
    }

    fun btTestarComponentesOnClick(view: View) {
        if(rbMasculino.isChecked){
            val snackbar = Snackbar.make(main, "Masculino Selecionado", Snackbar.LENGTH_INDEFINITE)
            snackbar.setBackgroundTint(Color.RED)
            snackbar.setAction("Cancelar"){
                snackbar.dismiss()
            }

            snackbar.show()
        }
    }
}